### DISCUSSION TITLE
Softwareserial RX not working in pins E0 (SWC) and E2 (SWD) ?

### DISCUSSION THREAD

**Discussion by hmeijdam** [dbuezas/lgt8fx#121] [Q&A]
For my application it would be nice to use the SWD (E2) pin as a softwareserial pin. 

I can use the SWD pin as analog input (A7) and as digital pin (23 or E2), and as Softwareserial TX, but I cannot use it for softwareserial RX.

Is this a known limitation for softwareserial RX?
I saw in the notes in the softwareserial example sketch that e.g. on a Leonardo and Mega not all pins support pinchange interrupt and therefore cannot be used as RX pin.

Hans

**Comment by LaZsolt** [comment]
I think I found a possibile solution. If the SWD and the SWC is disabled before initializing Software serial, then it may could work.
Disabling SWD and SWC, must write MCUSR twice within 4 clocks:
```
uint8_t newsr = MCUSR | _BV(SWDD);
MCUSR = newsr;
MCUSR = newsr;
```
![kép](https://user-images.githubusercontent.com/60486174/113504857-52d17580-953b-11eb-9e45-d208881989ee.png)

**Comment by youxiaojie** [comment]
by the way, does this two pin use standard jtag protocol?

**Reply by SuperUserNameMan** [reply]
Maybe you could help us find the datasheet ?

I tried using Baidu once, but since I don't read/write chinese, I failed.

---

Also, this [project]( https://github.com/SuperUserNameMan/LGTISP ) ( forked from [ this project ]( https://github.com/brother-yan/LGTISP ) ) contains a software implementation of the protocol.

Maybe it could give you some clues ?

**Reply by youxiaojie** [reply]
the datasheet is chinese, only one translated:( maybe you have.

**Reply by SuperUserNameMan** [reply]
> the datasheet is chinese, only one translated:( maybe you have.

We can use google-translate. It is ok.

We already have these datasheet :
- LGT8F328p v1.04 and v1.05
- LGT8F328d v1.0.5
- LGT8F88a v1.1.1

But none of them contain information about the SWD protocol ( " SWD双线片上调试/量产接口 " / " Serial Wire on-chip Debug(SWD) " ).

Maybe it is part of an other chinese datasheet ?

**Reply by youxiaojie** [reply]
but i see there is homemade arduino based isp, how did they achieve this?

**Reply by SuperUserNameMan** [reply]
the original arduino based ISP seems to come from LGT : https://github.com/LGTMCU/LarduinoISP
